packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -199,6 +199,14 @@ if (__DEV__) {
       return;
     }
 
+    // Don't warn about react-lifecycles-compat polyfilled components
+    if (
+      typeof instance.componentWillMount === 'function' &&
+      instance.componentWillMount.__suppressDeprecationWarning === true
```

Comment:
I expect that people will use this hidden flag to "hide" valid deprecations on components. Do we care about this use case? Probably not; I'm just noting that for people who do that, it would be counter-intuitive that we only check for that flag on `componentWillMount` but not on other methods here.

Reply:
> I expect that people will use this secret flag to fool React and "hide" valid deprecations on components. Do we care about this use case?

No, not really. I believe this was the best of all options considered, despite that minor drawback.

I only check that one method because it's all that's required to verify the presence of the polyfill.